kolibri/plugins/user/assets/src/views/sign-in-page/index.vue

```diff
@@ -68,30 +89,103 @@
       'icon-button': require('kolibri.coreVue.components.iconButton'),
       'core-textbox': require('kolibri.coreVue.components.textbox'),
       'logo': require('kolibri.coreVue.components.logo'),
+      'ui-autocomplete-suggestion': require('keen-ui/src/UiAutocompleteSuggestion'),
     },
     data: () => ({
       username: '',
       password: '',
+      usernameSuggestions: [],
+      suggestionTerm: '',
+      showDropdown: true,
+      highlightedIndex: -1,
     }),
+    watch: {
+      username: 'setSuggestionTerm',
+    },
     computed: {
       signUp() {
         return { name: PageNames.SIGN_UP };
       },
       versionMsg() {
         return this.$tr('poweredBy', { version: __version }); // eslint-disable-line no-undef
       },
+      canSignUp() {
+        return this.facilityConfig.learnerCanSignUp;
+      },
+      simpleLogin() {
+        return this.facilityConfig.learnerCanLoginWithNoPassword;
+      },
+      suggestions() {
+        // Filter suggestions on the client side so we don't hammer the server
+        return this.usernameSuggestions.filter(sug => sug.startsWith(this.username));
+      },
+      uniqueMatch() {
+        // If we have a matching username entered, don't show any suggestions.
+        return this.suggestions.length === 1 && this.suggestions[0] === this.username;
+      }
     },
     methods: {
+      handleKeyboardNav(e) {
+        switch (e.code) {
+          case 'ArrowDown':
+            this.highlightedIndex = Math.min(this.highlightedIndex + 1, this.suggestions.length - 1);
+            break;
+          case 'Enter':
+            this.fillUsername(this.suggestions[this.highlightedIndex]);
```

Comment:
We do already, see the `keydown.enter` event handler in the template.

The only reason I had to manually capture this in this way is because the keydown events are getting swallowed by the textbox KeenUI component we are using at the lowest level, and it is only emitting the event as a Vue event, not a native event.